ios/FluentUI/Table View/TableViewCellFileAccessoryView.swift

```diff
@@ -0,0 +1,383 @@
+            updateSharedStatus()
+        }
+    }
+
+    /// Actions to display in the accessory view's third column.
+    /// Depending on the cell's width, some actions may be hidden.
+    /// A maximum of 4 actions will be displayed.
+    /// Actions with the smallest index in the array will have a higher priority for being displayed.
+    @objc public var actions: [FileAccessoryViewAction] = [] {
+        didSet {
+            if tableViewCell != nil && !oldValue.elementsEqual(actions) {
+                updateLayout()
+            }
+        }
+    }
+
+    @objc public weak var tableViewCell: TableViewCell? {
+        didSet {
+            updateLayout()
+        }
+    }
+
+    @objc public override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        addSubview(columnStackView)
+
+        NSLayoutConstraint.activate([
+            columnStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            columnStackView.topAnchor.constraint(equalTo: topAnchor),
+            columnStackView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+
+        columnStackView.addArrangedSubview(actionsStackView)
+
+        updateSharedStatus()
+    }
+
+    @available(*, unavailable)
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if let tableViewCell = tableViewCell {
+            if !tableViewCell.bounds.equalTo(previousBounds) {
+                previousBounds = tableViewCell.bounds
+
+                updateLayout()
+            }
+        }
+    }
+
+    private struct Constants {
+        static let layoutBreakPoints: [CGFloat] = [424.0, 504.0, 584.0, 616.0, 752.0, 900.0, 1092.0]
+        static let maxVisibleActionCount: UInt8 = 4
+        static let minimumViewHeight: CGFloat = 24.0
+        static let actionsSpacingDefault: CGFloat = 16.0
+        static let actionsSpacingLarge: CGFloat = 24.0
+        static let columnSpacing: CGFloat = 24.0
+        static let reservedCellSpace: [CGFloat] = [460.0, 600.0]
+        static let columnMinWidth: CGFloat = 150.0
+        static let fullDateMinWidth: CGFloat = 170.0
+        static let sharedIconSize: CGFloat = 24.0
+        static let sharedStatusSpacing: CGFloat = 8.0
+    }
+
+    private lazy var actionsStackView: UIStackView = {
+        return createHorizontalStackView()
+    }()
+
+    private lazy var columnStackView: UIStackView = {
+        let columnStackView = createHorizontalStackView()
+        columnStackView.spacing = Constants.columnSpacing
+
+        return columnStackView
+    }()
+
+    private func createHorizontalStackView() -> UIStackView {
+        let stackView = UIStackView(frame: .zero)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+
+        return stackView
+    }
+
+    private lazy var dateLabel: Label = {
```

Comment:
Unfortunately no, this was a very important consideration for my approach. I explicitly did not want to build something too generic because this current implementation makes many assumptions about the content of the views to decide how to size them. I think the current implementation has a good balance in terms of flexibility because each column can be hidden/shown. We can assume that to represent a file/folder, a date column and a shared column will very often be used and we probably won't need to support more columns in the future.